scipy/optimize/tests/test_linprog.py

```diff
@@ -262,7 +264,7 @@ def test_linprog_cyclic_bland_bug_8561(self):
                           options=dict(maxiter=100, bland=True),
```

Comment:
Like this one there are quite a few places where the options are explicitly set for the simplex method. Would  changing these to include the `bland` kwarg in addition to set options?

Reply:
I'm not sure I understand the comment. I'll try to respond, though.

I don't think the `bland` option needs to be added here or anywhere it is not already. In general, setting options explicitly for the simplex method is OK (rather than adding them to a copy of `self.options`, which is typically done for interior-point) because the default options for simplex is an empty dictionary (see the `TestLinprogSimplex class`).

For the interior-point method, however, I wanted most of the unit tests to run for several different sparsity options. To accomplish that, I have a class for each for the different sparsity options, each class has its own`options` field (e.g. `options = {"sparse": True}` for the `TestLinprogIPSparse` class), and in the base class containing most of the test functions (`LinprogCommonTests`) there is `options=self.options` for most tests. The point of this PR was mostly to fix the places where I forgot to do that.

When I need to change an option for an `interior-point` test besides the sparsity options, typically I should be copying the options dictionary and adding the new option to that. For line 477, however, I suppose I decided that the test didn't need to run with the different sparsity options; running with the default (dense) option and without presolve was fine. To achieve this I just manually overrode the options structure. This was sort of lazy, as it means that the test will run several times (once for each subclass) but with the same options. 

So you're right, I probably should have fixed that here, too. I just didn't see it before. 

I suppose the rule is that manually setting options for simplex is OK, but for interior-point additional options should be added to a copy of self.options. If you see other places where that is not currently the case, let me know.

Reply:
You seem to have understood what I meant to say quite well. The idea of explicitly updating the options for the simplex method's tests stems from trying to set a consistent structure between the interior-point and simplex method. 

I've read up on the numerical stability / conditioning of the simplex method which I think causes the existing errors (I'll submit a more detailed issue later). It will also be good to have this set up when I make changes to the simplex method, rather than having to think of fixing a similar issue twice.

Reply:
@Kai-Striega Great to hear. Looking forward to reviewing them.